source/Engine.cpp

```diff
@@ -2322,15 +2345,34 @@ void Engine::AddSprites(const Ship &ship)
 {
 	bool hasFighters = ship.PositionFighters();
 	double cloak = ship.Cloaking();
+	bool damageHighlight = Preferences::Has("Damage highlights");
 	bool drawCloaked = (cloak && ship.IsYours());
+	bool drawHeat = ship.Heat() > 0.9;
+	bool drawShield = ship.RecentShieldDamage() > 4.;
+	double shield = sqrt(ship.RecentShieldDamage() / ship.Attributes().Get("shields"));
```

Comment:
It might be cheaper to add a field to `Ship` that can be used to store this, and set this value there (via a public setter method) instead of recalculating it later in `Engine::Draw()`.